File: ProjectEuler/5.py
LIMIT = 20

def solve(limit = LIMIT):
    from em import gcd
    
    total = 1
    for i in range(2, limit + 1):
        total = total * i // gcd(total, i)
    return total

if __name__ == "__main__":
	print(solve())

File: ProjectEuler/rust/solutions/src/bin/5.rs
use em::traits::Integer;

const LIMIT: u64 = 20;

fn solve(limit: Option<u64>) -> u64 {
    let limit = limit.unwrap_or(LIMIT);

    let mut total = 1;
    for i in 2..limit + 1 {
        total = total * i / total.gcd(&i);
    }
    total
}

fn main() {
    println!("{}", solve(None));
}
